language_detector/main.py

```diff
@@ -1,9 +1,35 @@
 """This is the entry point of the program."""
+# coding=utf-8
 
 from .languages import LANGUAGES
 
-
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    text = text.strip().split("\n\n")
+    count = {}
+    
+    for t in text:
+        count[find_lang(t, languages=LANGUAGES)] = len(t)
+        
+    return max(count, key=count.get)
+
+
+def find_lang(text, languages=LANGUAGES):
+    for ch in "-(),.":
+        text = text.replace(ch, "")
+        
+    text = text.strip().split(" ")
+    text = [el.strip() for el in text if el != ""]
+
+    wordcount = {}
+    for word in text:
+        if word not in wordcount:
```

Comment:
This can be simplified by using the setdefault function:
wordcount.setdefault(word, 0) += 1
Alternatively you could use a defaultdict or a Counter instead of a normal dictionary (from the collections module).
All of these basicly create a standart value for every entry so you don't need to check if the word is in there already.